python/python/lance/util.py

```diff
@@ -40,3 +45,71 @@ def sanitize_ts(ts: ts_types) -> datetime:
     elif not isinstance(ts, datetime):
         raise TypeError(f"Unrecognized version timestamp {ts} of type {type(ts)}")
     return ts
+
+
+class KMeans:
+    def __init__(self, k: int, metric_type: str = "l2", max_iters: int = 50):
+        """Create a KMeans model.
```

Comment:
What approach do we use to determine the starting centroids?  Random selection of points or do we take the first k data points?  Should we explain this here or hide this detail?

Reply:
This is just random initialization. I will add comment here. We do support kmeans++ internally, but did not expose the choice.

Reply:
Added comments. we only use random initialization now. `kmeans++` was implement in rust but not exposed yet.